### PR TITLE
[MM-26459] Split logs for agent and coordinator

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR splits logging for the two main components. 
This is needed to avoid having a single `ltapi.log` file with everything inside.

#### Ticket

https://mattermost.atlassian.net/browse/MM-26459

### CODE REVIEW

File: logger/logger.go [hunk 1]
```diff
@@ -19,6 +19,18 @@ type Settings struct {
 	FileLocation  string `default:"loadtest.log"`
 }
 
+func New(logSettings *Settings) *mlog.Logger {
```
Comment (agnivade) [inline logger/logger.go:22]: godoc please :)

File: api/server.go [hunk 2]
```diff
@@ -12,13 +12,16 @@ import (
...
+	"github.com/mattermost/mattermost-server/v5/mlog"
 )
 
 // api keeps track of the load-test API server state.
 type api struct {
 	agents       map[string]*loadtest.LoadTester
 	coordinators map[string]*coordinator.Coordinator
 	metrics      *performance.Metrics
+	clog         *mlog.Logger
+	alog         *mlog.Logger
```
Comment (agnivade) [inline api/server.go:24]: Would prefer them to be `coordLog` and `agentLog`. Somebody reading the codebase without any context would not understand quickly what they mean.

File: cmd/ltapi/main.go [hunk 3]
```diff
@@ -18,6 +18,7 @@ import (
 func RunServerCmdF(cmd *cobra.Command, args []string) error {
 	port, _ := cmd.Flags().GetInt("port")
 
+	// TODO: add a config file for the API server.
```
Comment (agnivade) [inline cmd/ltapi/main.go:21]: This is getting really out of hand now with so many config files :)

Comment (streamer45) [reply cmd/ltapi/main.go:21]: Well, I am still not convinced by the monolithic config approach :)

File: api/server.go (hunk 2 above)
Comment (streamer45) [reply api/server.go:24]: Fair enough :)

File: coordinator/cluster/cluster.go [hunk 4]
```diff
@@ -29,14 +30,17 @@ type errorTrack struct {
 
 // New creates and initializes a new LoadAgentCluster for the given config.
 // An error is returned if the initialization fails.
-func New(config LoadAgentClusterConfig) (*LoadAgentCluster, error) {
+func New(config LoadAgentClusterConfig, log *mlog.Logger) (*LoadAgentCluster, error) {
+	if log == nil {
+		return nil, fmt.Errorf("logger should not be nil")
```
[the remark this answers is not in the document]
Comment (streamer45) [reply coordinator/cluster/cluster.go:35]: Yes, my reasoning there was to use `fmt` if the `errors` package was not already imported. It felt weird to import one package for a single log line. Also, [MM-23914](https://mattermost.atlassian.net/browse/MM-23914) is around the corner.

**Pull request merged**